pkg/core/resources/apis/mesh/resources.go

```diff
@@ -110,6 +110,96 @@ func init() {
 	registry.RegistryListType(&CircuitBreakerResourceList{})
 }
 
+const (
+	DataplaneType model.ResourceType = "Dataplane"
+)
+
+var _ model.Resource = &DataplaneResource{}
+
+type DataplaneResource struct {
+	Meta model.ResourceMeta
+	Spec *mesh_proto.Dataplane
+}
+
+func NewDataplaneResource() *DataplaneResource {
+	return &DataplaneResource{
+		Spec: &mesh_proto.Dataplane{},
+	}
+}
+
+func (t *DataplaneResource) GetType() model.ResourceType {
+	return DataplaneType
+}
+
+func (t *DataplaneResource) GetMeta() model.ResourceMeta {
+	return t.Meta
+}
+
+func (t *DataplaneResource) SetMeta(m model.ResourceMeta) {
+	t.Meta = m
+}
+
+func (t *DataplaneResource) GetSpec() model.ResourceSpec {
+	return t.Spec
+}
+
+func (t *DataplaneResource) SetSpec(spec model.ResourceSpec) error {
+	protoType, ok := spec.(*mesh_proto.Dataplane)
+	if !ok {
+		return fmt.Errorf("invalid type %T for Spec", spec)
+	} else {
+		t.Spec = protoType
+		return nil
+	}
+}
+
+func (t *DataplaneResource) Scope() model.ResourceScope {
+
```

Comment:
oh btw it seems like the template should be fixed to get rid of these empty lines

Reply:
Yeh it's just hard to get Go templates to have exactly the whitespace you need :)